olp-cpp-sdk-core/src/cache/InMemoryCache.cpp

```diff
@@ -123,42 +107,48 @@ void InMemoryCache::RemoveKeysWithPrefix(const std::string& keyPrefix) {
 
 bool InMemoryCache::PurgeExpired() {
   bool ret = true;
-  std::vector<time_t> expiredKeys;
-  for (auto item : m_itemExpiries) {
-    if (item.first < m_timeProvider())
-      expiredKeys.push_back(item.first);
-    else
-      break;  // std::map is sorted, so any further expiry times will be larger.
+  std::vector<time_t> expired_keys;
+  const auto time_now = time_provider_();
+
+  for (const auto& item : item_expiries_) {
+    if (item.first < time_now) {
+      expired_keys.push_back(item.first);
+    } else {
+      // std::map is sorted, so any further expiry times will be larger.
+      break;
+    }
   }
 
-  for (auto key : expiredKeys) ret &= PurgeExpired(key);
+  for (auto& key : expired_keys) {
+    ret &= PurgeExpired(key);
+  }
 
   return ret;
```

Comment:
Seems like `return true;` maybe used instead.

Reply:
It returns only true if all PurgeExpired returns true.

Reply:
InMemoryCache::PurgeExpired(time_t expire_time) returns true unconditionally